starter-code/src/clue.js

```diff
@@ -1,94 +1,126 @@
+ocupation:   "Scientist"
+};
+
+var profPlum = {
+first_name:   "Victor",
+last_name:    "Plum",
+color:        "purple",
+description:  "Billionare video game designer",
+age:          22,
+image:        "https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg",
+occupation:   "Designer"
+};
+
+var missScarlet = {
+first_name:   "Kasandra",
+last_name:    "Scarlet",
+color:        "red",
+description:  "She is an A-list movie star with a dark past",
+age:          31,
+image:        "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+occupation:   "Actor"
+};
+
+var mrsPeacock = {
+first_name:   "Eleanor",
+last_name:    "Peacock",
+color:        "blue",
+description:  "She is from a wealthy family and uses her status and money to earn popularity",
+age:          36,
+image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+occupation:   "Socialité"
+};
+
+var mrMustard = {
+first_name:   "Jack",
+last_name:    "Mustard",
+color:        "yellow",
+description:  "He is a former football player who tries to get by on his former glory",
+age:          62,
+image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+occupation:   "Retired Football player"
+};
+
+
+
+//weapons
+
+var rope = {name: "rope", weight: 10};
+var knife = {name: "knife", weight: 8};
+var candlestick = {name: "candlestick", weight: 2};
+var dumbbell = {name: "dumbbell", weight: 30};
+var poison = {name: "poison", weight: 2};
+var axe = {name: "axe", weight: 15};
+var bat = {name: "bat", weight: 13};
+var trophy = {name: "trophy", weight: 25};
+var pistol = {name: "pistol", weight: 20};
+
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+
+
+var dinningRoom = {name: "Dinning Room"}
+var conservatory = {name: "Conservatory"};
+var kitchen = {name: "Kitchen"};
+var study = {name: "Study"};
+var library = {name: "Library"};
+var billiardRoom = {name: "Billiard Room"};
+var lounge = {name: "Lounge"};
+var ballroom = {name: "Ballroom"};
+var hall = {name: "Hall"};
+var spa = {name: "Spa"};
+var livingRoom = {name: "Living Room"};
+var observatory = {name: "Observatory"};
+var theater = {name: "Theater"};
+var guestHouse = {name: "Guest House"};
+var patio = {name: "Patio"};
+
 
 // Characters Collection
-var charactersArray = [];
+var charactersArray = [mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard];
 
 // Rooms' Collection
-var roomsArray = [];
+var roomsArray = [dinningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio];
 
 // Weapons Collection
-var weaponsArray = [];
+var weaponsArray = [rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol];
+
+function randomSelector(array) {
```

Comment:
You don't need to add "+1" as you are in an array, and the index only go from 0 to 9, the right code should be 👍 
function randomSelector(array) {
  var nb = Math.floor(Math.random() * array.length);
  return array[nb];
}


Because if you adding one it will sometimes check the index [10] and as there is nothing on index 10 it will return undefined